routing/pathfind.go

```diff
@@ -1243,22 +1309,26 @@ func findBlindedPaths(g Graph, target route.Vertex,
 	// a node that doesn't have any other edges - in that final case, the
 	// whole path should be ignored.
 	paths, _, err := processNodeForBlindedPath(
-		g, target, supportsRouteBlinding, nil, restrictions,
+		g, nextTarget, supportsRouteBlinding, visited, restrictions,
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	// Reverse each path so that the order is correct (from introduction
-	// node to last hop node) and then append this node on as the
-	// destination of each path.
 	orderedPaths := make([][]blindedHop, len(paths))
-	for i, path := range paths {
-		sort.Slice(path, func(i, j int) bool {
-			return j < i
-		})
+	if len(paths) == 0 && len(restrictions.incomingChainedChannels) > 0 {
```

Comment:
Missing comment here why we have this exception. Moreover I think the below check now :

```
if restrictions.minNumHops == 0 {
		singleHopPath := [][]blindedHop{{{vertex: target}}}
```

just works because we check that the minHopNumber is equal to the incoming channel, which means we won't add the target here in case we have a chainedList. Maybe add a comment or also add a check here?